DriveFusion/ErrorSignaling.h

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2015 Google Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+#pragma once
+
+#define CHECK_ARG(condition) \
+    do \
+    { \
+      if (!(condition)) \
+      { \
+        hr = E_INVALIDARG; \
+        Log::WriteOutput( \
+            LogType::Error, L"\tArgument failed condition " L#condition); \
+        return hr; \
+      } \
+    } while (false)
+
+#define CHECK_HR(function) \
```

Comment:
I'd rename the function argument as something like hresult, since it's actually just the result code that the function returned.